crates/db-device/src/device_manager.rs

```diff
@@ -0,0 +1,592 @@
+// Copyright 2020 Alibaba Cloud. All Rights Reserved.
+// Copyright © 2019 Intel Corporation. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! System level device management.
+//!
+//! [IoManager](struct.IoManager.html) is respondsible for managing
+//! all devices of virtual machine, registering IO resources callback,
+//! unregistering devices and helping VM IO exit handling.
+//！
+//！VMM would be responsible for getting device resource request, ask
+//! vm_allocator to allocate the resources, ask vm_device to register the
+//! devices IO ranges, and finally set resources to virtual device.
+
+use std::cmp::{Ord, Ordering, PartialEq, PartialOrd};
+use std::collections::btree_map::BTreeMap;
+use std::fmt::{Display, Formatter};
+use std::ops::Deref;
+use std::result;
+use std::sync::Arc;
+
+use crate::resources::Resource;
+#[cfg(any(target_arch = "x86_64", target_arch = "x86"))]
```

Comment:
We only support x86_64 in db-arch, so in other crates the rule should stay the same.

#[cfg(target_arch = "x86_64")] is enough.